test/simplegraphs/simplegraphs.jl

```diff
@@ -4,7 +4,7 @@ import Random
     adjmx1 = [0 1 0; 1 0 1; 0 1 0] # graph
     adjmx2 = [0 1 0; 1 0 1; 1 1 0] # digraph
     # specific concrete generators - no need for loop
-    @test @inferred(eltype(SimpleGraph())) == Int
+    @test @inferred(eltype(SimpleGraph)) == Int
```

Comment:
`SimpleGraph` is not a type, but a convenience constructor for `SimpleGraph{Int64}`, so `eltype` is not changed for this, causing test to fail. Maybe we should define `SimpleGraph = SimpleGraph{Int64}` instead of just defining the constructor...

Reply:
Indeed, I would also support `Any`. One could maybe argue for `Integer` instead of `Any` though...

The issue here is, that we have defined `eltype(::Type{AbstractGraph}) = _NI("eltype")` , a decision which I am not sure if I like. So maybe we should just check here if an error is being thrown. Or we just delete the test at all, as we are only concerned with concrete subtypes of `AbstractGraph`.

Reply:
I think we should do
```julia
eltype(::Type{AbstractGraph{T}) where T = T
```
as @simonschoelly suggested above